rewindable-index/src/RewindableIndex/Storable.hs

```diff
@@ -189,16 +193,31 @@ newtype Config = Config
   } deriving (Show, Eq)
 $(Lens.makeLenses ''Config)
 
+data SyntheticEvent e p =
```

Comment:
Haddock comment here. What is that used for and what do the `e` and `p` typically represent. Your PR description is a good starting point as it describes a lot of this :)

Reply:
I added a comment on the `checkpoint` function. In theory this `SyntheticEvent` should not be shared.

Reply:
Ah, alright